src/middlewared/middlewared/plugins/reporting/ifstat.py

```diff
@@ -0,0 +1,89 @@
+from pyroute2.ethtool import Ethtool
+from pyroute2.ethtool.ioctl import NotSupportedError, NoSuchDevice
+from psutil import net_if_stats, net_io_counters
+
+
+class IfStats(object):
+
+    def __init__(self, interval, prev_data, ignore_ifaces):
+        self.interval = interval
+        self.prev_data = prev_data
+        self.ignore = ignore_ifaces
+        self.eth = Ethtool()
+
+    def __enter__(self):
+        return self.read()
+
+    def __exit__(self, _type, _value, _tb):
+        self.eth.close()
```

Comment:
I am not sure what `open`/`close` does, but can it have a little performance impact to keep this open during the whole `reporting.realtime` lifetime?

Reply:
Nothing too measurable from what I could see. `Ethtool()` is opening a standard socket so it's closing it on exit.

Reply:
However, I am going to investigate that a little more to see if I can make the class instantiation persistent in a later PR.